src/js/components/uploadDetachedFiles/uploadDetachedFileMeta.jsx

```diff
@@ -0,0 +1,458 @@
+				};
+				dFile.valid = false;
+			}
+			else {
+				// valid!
+				dFile.error = {
+					show: false,
+					header: '',
+					description: ''
+				};
+				dFile.valid = true;
+			}
+		}
+		else {
+			// not all dates exist yet
+			dFile.error = {
+				show: false,
+				header: '',
+				description: ''
+			};
+			dFile.valid = false;
+		}
+
+		this.setState({
+			[file]:dFile,
+			showUploadFilesBox: dFile.valid
+		});
+	}
+
+	showError(file, header, description) {
+		// Show any error that occurs at any point during file upload
+		const state = Object.assign({}, this.state[file], {
+			error: {
+				show: true,
+				header: header,
+				description: description
+			}
+		});
+		
+		this.setState({
+			[file]: state
+		});
+	}
+
+	hideError(file) {
+		// Stop displaying the error for the given file
+		const state = Object.assign({}, this.state[file], {
+			error: {
+				show: false,
+				header: '',
+				description: ''
+			}
+		});
+
+		this.setState({
+			[file]: state
+		});
+	}
+
+	uploadFile() {
+		// upload specified file
+		this.props.setSubmissionState('uploading');
+		this.props.submission.meta['startDate'] = this.state.detachedAward.startDate.format('DD/MM/YYYY');
+		this.props.submission.meta['endDate'] = this.state.detachedAward.endDate.format('DD/MM/YYYY');
+		this.props.submission.meta['subTierAgency'] = this.state.agency;
+		
+		if (kGlobalConstants.LOCAL == true && !this.isUnmounted) {
+			UploadHelper.performDetachedLocalUpload(this.props.submission)
+                .then((submissionID) => {
+                    // TODO: Remove this when this is eventually tied to user accounts
+                    this.props.setSubmissionId(submissionID);
+					this.checkFileStatus(submissionID);
+					this.props.history.push('/uploaddetachedfiles/'+submissionID);
+					this.props.validate();
+                })
+				.catch((err) => {
+					if (err.httpStatus == 403) {
+						this.setState({
+							notAllowed: true,
+							errorMessage: err.message
+						});
+					}
+					else {
+						this.setState({
+							errorMessage: err.body.message
+						});
+					}
+				});
+		}
+		else {
+			UploadHelper.performDetachedFileUpload(this.props.submission)
+				.then((submissionID) => {
+					// TODO: Remove this when this is eventually tied to user accounts
+					this.props.setSubmissionId(submissionID);
+					this.checkFileStatus(submissionID);
+					this.props.history.push('/uploaddetachedfiles/'+submissionID);
+					this.props.validate();
+
+				})
+				.catch((err) => {
+					if (err.httpStatus == 403) {
+						this.setState({
+							notAllowed: true,
+							errorMessage: err.message
+						});
+					}
+					else {
+						this.setState({
+							errorMessage: err.message
+						});
+					}
+				});
+		}
+	}
+
+	checkFileStatus(submissionID) {
+		// callback to check file status
+		GenerateFilesHelper.fetchSubmissionMetadata(submissionID)
+			.then((response) => {
+				if (this.isUnmounted) {
+					return;
+				}
+				var submission = true;
```

Comment:
it was from before when meta and validation were all located in page. no longer needed so will be removed